pkg/clusterversion/cockroach_versions.go

```diff
@@ -549,29 +549,18 @@ const (
 	finalVersion = invalidVersionKey
 )
 
-// devVersionsAbove is the version key above which all versions are offset to be
-// development version when developmentBranch is true. By default this is all
-// versions, by setting this to -1, but an env var can override this, to leave
-// the first version un-offset. Doing so means that that version, which is
-// generally minBinaryVersion as well, is unchanged, and thus allows upgrading a
-// stable release data-dir to a dev version if desired.
-var devVersionsAbove Key = func() Key {
-	if envutil.EnvOrDefaultBool("COCKROACH_UPGRADE_TO_DEV_VERSION", false) {
-		return invalidVersionKey + 1
-	}
-	return invalidVersionKey
-}()
-
 var versionsSingleton = func() keyedVersions {
 	if developmentBranch {
+		skipFirst := envutil.EnvOrDefaultBool("COCKROACH_UPGRADE_TO_DEV_VERSION", false)
```

Comment:
the deleted code sort of explained what the environment variable did. from reading the new code, i don't find it that obvious. could we have a comment here that gives an explicit definition of the behavior of this env var?

Reply:
oh i see that it's documented more on line 543. can we consolidate the `envutil.EnvOrDefaultBool("COCKROACH_UPGRADE_TO_DEV_VERSION", false)` into a single variable that is reused in both places?

Reply:
These are different variables, aren't they? This this the opt-in to doing a dev upgrade and that one is the force-non-dev-to-present-as-dev isn't it?

Reply:
fair enough; I don't think the removed section really explained all that much, but I added a bigger comment that hopefully does this time.